llvm/test/CodeGen/X86/apx/push2-pop2-cfi-seh.ll

```diff
@@ -0,0 +1,217 @@
+;
+; WIN-REF-LABEL: csr6_alloc16:
+; WIN-REF:       # %bb.0: # %entry
+; WIN-REF-NEXT:    pushq %r15
+; WIN-REF-NEXT:    .seh_pushreg %r15
+; WIN-REF-NEXT:    pushq %r14
+; WIN-REF-NEXT:    .seh_pushreg %r14
+; WIN-REF-NEXT:    pushq %r13
+; WIN-REF-NEXT:    .seh_pushreg %r13
+; WIN-REF-NEXT:    pushq %r12
+; WIN-REF-NEXT:    .seh_pushreg %r12
+; WIN-REF-NEXT:    pushq %rbp
+; WIN-REF-NEXT:    .seh_pushreg %rbp
+; WIN-REF-NEXT:    pushq %rbx
+; WIN-REF-NEXT:    .seh_pushreg %rbx
+; WIN-REF-NEXT:    subq $56, %rsp
+; WIN-REF-NEXT:    .seh_stackalloc 56
+; WIN-REF-NEXT:    .seh_endprologue
+; WIN-REF-NEXT:    #APP
+; WIN-REF-NEXT:    #NO_APP
+; WIN-REF-NEXT:    xorl %eax, %eax
+; WIN-REF-NEXT:    callq *%rax
+; WIN-REF-NEXT:    nop
+; WIN-REF-NEXT:    addq $56, %rsp
+; WIN-REF-NEXT:    popq %rbx
+; WIN-REF-NEXT:    popq %rbp
+; WIN-REF-NEXT:    popq %r12
+; WIN-REF-NEXT:    popq %r13
+; WIN-REF-NEXT:    popq %r14
+; WIN-REF-NEXT:    popq %r15
+; WIN-REF-NEXT:    retq
+; WIN-REF-NEXT:    .seh_endproc
+;
+; WIN-LABEL: csr6_alloc16:
+; WIN:       # %bb.0: # %entry
+; WIN-NEXT:    pushq %rax
+; WIN-NEXT:    .seh_pushreg %rax
+; WIN-NEXT:    push2 %r14, %r15
+; WIN-NEXT:    .seh_pushreg %r15
+; WIN-NEXT:    .seh_pushreg %r14
+; WIN-NEXT:    push2 %r12, %r13
+; WIN-NEXT:    .seh_pushreg %r13
+; WIN-NEXT:    .seh_pushreg %r12
+; WIN-NEXT:    push2 %rbx, %rbp
+; WIN-NEXT:    .seh_pushreg %rbp
+; WIN-NEXT:    .seh_pushreg %rbx
+; WIN-NEXT:    subq $48, %rsp
+; WIN-NEXT:    .seh_stackalloc 48
+; WIN-NEXT:    .seh_endprologue
+; WIN-NEXT:    #APP
+; WIN-NEXT:    #NO_APP
+; WIN-NEXT:    xorl %eax, %eax
+; WIN-NEXT:    callq *%rax
+; WIN-NEXT:    nop
+; WIN-NEXT:    addq $48, %rsp
+; WIN-NEXT:    pop2 %rbp, %rbx
+; WIN-NEXT:    pop2 %r13, %r12
+; WIN-NEXT:    pop2 %r15, %r14
+; WIN-NEXT:    popq %rcx
+; WIN-NEXT:    retq
+; WIN-NEXT:    .seh_endproc
+;
+; WIN-PPX-LABEL: csr6_alloc16:
+; WIN-PPX:       # %bb.0: # %entry
+; WIN-PPX-NEXT:    pushq %rax
+; WIN-PPX-NEXT:    .seh_pushreg %rax
+; WIN-PPX-NEXT:    push2p %r14, %r15
+; WIN-PPX-NEXT:    .seh_pushreg %r15
+; WIN-PPX-NEXT:    .seh_pushreg %r14
+; WIN-PPX-NEXT:    push2p %r12, %r13
+; WIN-PPX-NEXT:    .seh_pushreg %r13
+; WIN-PPX-NEXT:    .seh_pushreg %r12
+; WIN-PPX-NEXT:    push2p %rbx, %rbp
+; WIN-PPX-NEXT:    .seh_pushreg %rbp
+; WIN-PPX-NEXT:    .seh_pushreg %rbx
+; WIN-PPX-NEXT:    subq $48, %rsp
+; WIN-PPX-NEXT:    .seh_stackalloc 48
+; WIN-PPX-NEXT:    .seh_endprologue
+; WIN-PPX-NEXT:    #APP
+; WIN-PPX-NEXT:    #NO_APP
+; WIN-PPX-NEXT:    xorl %eax, %eax
+; WIN-PPX-NEXT:    callq *%rax
+; WIN-PPX-NEXT:    nop
+; WIN-PPX-NEXT:    addq $48, %rsp
+; WIN-PPX-NEXT:    pop2p %rbp, %rbx
+; WIN-PPX-NEXT:    pop2p %r13, %r12
+; WIN-PPX-NEXT:    pop2p %r15, %r14
+; WIN-PPX-NEXT:    popq %rcx
+; WIN-PPX-NEXT:    retq
+; WIN-PPX-NEXT:    .seh_endproc
+entry:
+  tail call void asm sideeffect "", "~{rbp},~{r15},~{r14},~{r13},~{r12},~{rbx},~{dirflag},~{fpsr},~{flags}"()
+  %a = alloca [2 x ptr], align 8
```

Comment:
What happens if we require [3 x ptr] here. Maybe change to 3 to cover this case.

Reply:
The change will look like
```
-; WIN-NEXT:    subq $48, %rsp
-; WIN-NEXT:    .seh_stackalloc 48
+; WIN-NEXT:    subq $64, %rsp
+; WIN-NEXT:    .seh_stackalloc 64
-; WIN-NEXT:    addq $48, %rsp
+; WIN-NEXT:    addq $64, %rsp
```
What do you want to check?

Reply:
This test is used to check to CFI & SEH for PP2/PPX. The CFI & SEH after the alloca should not matter.

Reply:
I see. Done (2->3)